compaction/compaction.cc

```diff
@@ -427,38 +427,10 @@ uint64_t compaction_progress_monitor::get_progress() const {
     return _progress;
 }
 
-class formatted_sstables_list {
-    bool _include_origin = true;
-    std::vector<std::string> _ssts;
-public:
-    formatted_sstables_list() = default;
-    void reserve(size_t n) {
-        _ssts.reserve(n);
-    }
-    explicit formatted_sstables_list(const std::vector<shared_sstable>& ssts, bool include_origin) : _include_origin(include_origin) {
-        _ssts.reserve(ssts.size());
-        for (const auto& sst : ssts) {
-            *this += sst;
-        }
-    }
-    formatted_sstables_list& operator+=(const shared_sstable& sst) {
-        _ssts.emplace_back(to_string(sst, _include_origin));
-        return *this;
-    }
-    friend fmt::formatter<formatted_sstables_list>;
-};
-
+auto format_sstables_list(const std::vector<shared_sstable>& list, bool with_origin) {
+    return fmt::join(list | boost::adaptors::transformed([with_origin] (auto sst) { return to_string(sst, with_origin); }), ", ");
 }
```

Comment:
Is this returning a reference to a temporary?

Reply:
It's also slower, as it's generating strings and then formatting them again.

Reply:
> Is this returning a reference to a temporary?

Shouldn't. `list` is available externally, `fmt::join` and `boost::adatptors::...` both return a value, as far as I can tell, `to_string` returns `std::string`, not view

Reply:
> It's also slower, as it's generating strings and then formatting them again.

Slower than existing code? It also generates strings, then formats them again. Or you mean that it could be better if not using to_string() at all?

Reply:
fmt::join() doesn't format, IIRC it's just a wrapper.

Reply:
> > Is this returning a reference to a temporary?
> 
> Shouldn't. `list` is available externally, `fmt::join` and `boost::adatptors::...` both return a value, as far as I can tell, `to_string` returns `std::string`, not view

Was trying to guess the cause of the test failures.